From 0c963383cd49460ebe22821753df9545676edc70 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 20 Apr 2020 00:11:08 +0300
Subject: [PATCH] [SW-0] upd_pending: switch to regmap

Remove the per platform getter functions providing the "update pending"
register offset.
This is in favor of the regmap headers available in specs.
This should lower the code size as the resolving is done in compile time
and no variables used to hold the offset.

Change-Id: I78b0fa0da605fdef59c90820f340cc3e4ee2ba07
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/app/ppboot.c b/app/ppboot.c
index 1dbcddf..cc4b961 100644
--- a/app/ppboot.c
+++ b/app/ppboot.c
@@ -48,8 +48,7 @@
 
 static enum fw_post_reset_cmd preboot_image_offset(u32 *offset)
 {
-	u32 upd_pending_reg = get_upd_pending_sts_reg();
-	u32 image_pending = regmap_readl(upd_pending_reg);
+	u32 image_pending = regmap_readl(mmUPD_PENDING_STS);
 	u32 off, idx;
 
 	if (boot_desc_init()) {
diff --git a/app/preboot.c b/app/preboot.c
index e00037a..7af436c 100644
--- a/app/preboot.c
+++ b/app/preboot.c
@@ -43,7 +43,6 @@
 static void preboot_set_upd_pending_reg(u32 modifiable_id)
 {
 	u32 magic;
-	u32 upd_pending_reg = get_upd_pending_sts_reg();
 
 	switch (modifiable_id) {
 	case 0:
@@ -56,7 +55,7 @@
 		magic = FLOPS_UPD_NOOP;
 	}
 
-	regmap_writel(magic, upd_pending_reg);
+	regmap_writel(magic, mmUPD_PENDING_STS);
 }
 
 static u32 modifiable_image_base_offset(bool *pending)
@@ -88,8 +87,7 @@
 
 static u32 next_image_offset(bool *pending)
 {
-	u32 upd_pending_reg = get_upd_pending_sts_reg();
-	u32 image_pending = regmap_readl(upd_pending_reg);
+	u32 image_pending = regmap_readl(mmUPD_PENDING_STS);
 	u32 offset, idx;
 
 	if (boot_desc_init()) {
diff --git a/gaudi/common.c b/gaudi/common.c
index 585ab60..931c8f4 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -190,10 +190,6 @@
 	return mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS;
 }
 
-u32 get_upd_pending_sts_reg(void)
-{
-	return mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_3;
-}
 /*
  * Wait for Boot Manager to complete PCIe image execution
  * and return to idle state, so it won't initiate UART access anymore
diff --git a/goya/common.c b/goya/common.c
index 373209c..95068d0 100644
--- a/goya/common.c
+++ b/goya/common.c
@@ -162,11 +162,6 @@
 	return mmPSOC_GLOBAL_CONF_WARM_REBOOT;
 }
 
-u32 get_upd_pending_sts_reg(void)
-{
-	return mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3;
-}
-
 void preboot_sync_btl(void) {}
 
 int hl_pcie_init_only(void)
diff --git a/goya2/common.c b/goya2/common.c
index c3559c2..75146ea 100644
--- a/goya2/common.c
+++ b/goya2/common.c
@@ -192,10 +192,6 @@
 	return mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS;
 }
 
-u32 get_upd_pending_sts_reg(void)
-{
-	return mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_3;
-}
 /*
  * Wait for Boot Manager to complete PCIe image execution
  * and return to idle state, so it won't initiate UART access anymore
diff --git a/goya2_arc_gnu/common.c b/goya2_arc_gnu/common.c
index 454451e..db634cd 100644
--- a/goya2_arc_gnu/common.c
+++ b/goya2_arc_gnu/common.c
@@ -176,11 +176,6 @@
 	return mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS;
 }
 
-u32 get_upd_pending_sts_reg(void)
-{
-	return mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3;
-}
-
 void preboot_sync_btl(void) {}
 
 
diff --git a/include/preboot.h b/include/preboot.h
index 28feb81..8f59326 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -75,7 +75,6 @@
 }
 
 u32 get_cpu_boot_status_reg(void);
-u32 get_upd_pending_sts_reg(void);
 u32 hl_skip_pcie_init(void);
 void hl_uart_init(void);
 u32 hl_spi_baudr_get(void);
